proposals/0200-raw-string-escaping-revision-1.md

```diff
@@ -0,0 +1,454 @@
+
+In Rust, raw string literals are written as `r"..."`. To embed double-quotes in a Rust raw string literal, you add one or more pound signs before the opening quote, and put a matching number of pound signs after the closing quote: `r#"..."..."#`, `r##"...#"..."##`, etc. Rust developers assured us that even one pound sign was unusual and more than one almost never needed but it's nice to have the flexibility in the rare cases where you need it.
+
+### Swiftifying Rust's Design
+
+Rust's design distinguishes between conventional and raw string literals. It also includes an asymmetric `r` off its leading edge. We found these distinctions unnecessary and the `r` aesthetically displeasing. Instead, our design powers up a conventional Swift `String` literal and in doing so, allows you to access features normally associated with raw and cooked literals.
+
+In this design, there is only *one* variety of string literals. There is no special "raw" syntax. A string literal is either 
+
+* a sequence of characters surrounded by double quotation marks ("), or 
+* a string that spans several lines surrounded by three double quotation marks.
+
+These are examples of Swift string literals:
+
+```
+"This is a single line Swift string literal"
+
+"""
+    This is a multi line
+    Swift string literal
+    """
+```
+
+In this form, the revised string design acts exactly like any other string. You use escape character sequences including string interpolation exactly as you would today. A backslash escape character tells the compiler that a sequence should be interpolated, interpreted as an escaped character, or represent a unicode scalar. 
+
+Swift's escape sequences include:
+
+* The special characters `\0` (null character), `\\` (backslash), `\t` (horizontal tab), `\n` (line feed), `\r` (carriage return), `\"` (double quotation mark) and `\'` (single quotation mark) 
+* Arbitrary Unicode scalars, written as `\u{n}`, where *n* is a 1–8 digit hexadecimal number with a value equal to a valid Unicode code point
+* Interpolated expressions, introduced by `\(` and terminated by `)`
+
+### Expanding Delimiters
+
+Our design adds customizable string delimiters. You may pad a string literal with one or more `#` (pound, Number Sign, U+0023) characters:
+
+```
+"This is a Swift string literal"
+
+#"This is also a Swift string literal"#
+
+####"So is this"####
+```
+
+The number of pound signs at the start of the string (in these examples, zero, one, and four) must match the number of pound signs at the end of the string. `"This"`, `#"This"#`, and `##"This"##` represent identical string values:
+
+```
+static-string-literal -> " quoted-text " |
+   """ multiline-quoted-text """ |
+   # static-string-literal #
+```
+
+Adding a pound signs *changes the string literal delimiter* allowing you to "cook" a string and include unescaped double quotes:
```

Comment:
I would rephrase this as:

```
Any instance of the delimiter which is not followed by the appropriate number of pound signs
is treated as literal string contents, rather than as the end of the string literal. That is,
the leading pound signs *change the string's delimiter* from `"` to `"#` (or `"##`', etc.).
A plain `"` without pound signs after it is just a double-quote character inside the string.
```